src/util/tprint.cpp

```diff
@@ -148,3 +156,24 @@ TEST_CASE("reuse sntprint buffer - with formatting over buf_size", {
     ASSERT(std::strcmp(buffer, "3") == 0, "Got '{}', '3' expected", buffer);
     return true;
 });
+
+TEST_CASE("sntprint - extra {}", {
+    char buffer[4] = {};
+    sntprint(buffer, LEN(buffer), "{}{}", 1);
+    ASSERT(std::strcmp(buffer, "1") == 0, "Got '{}', '1' expected", buffer);
+    return true;
+});
+
+TEST_CASE("sntprint - extra {}", {
+    char buffer[4] = {};
+    sntprint(buffer, LEN(buffer), "{}{}{}{}{}{}", 2);
+    ASSERT(std::strcmp(buffer, "2") == 0, "Got '{}', '2' expected", buffer);
+    return true;
+});
+
+TEST_CASE("sntprint - just enough", {
+    char buffer[10] = {};
+    sntprint(buffer, LEN(buffer), "{}-{}-{}%{", 3, 2, 1);
+    ASSERT(std::strcmp(buffer, "3-2-1") == 0, "Got '{}', '3-2-1{' expected", buffer);
```

Comment:
```suggestion
    ASSERT(std::strcmp(buffer, "3-2-1") == 0, "Got '{}', '3-2-1' expected", buffer);
```

Reply:
Dunno what's up with the syntax highlighter here, oof

Reply:
This will break the tests. I checked that escaped "{" on the end of the string are retained.

Reply:
That is what the industry calls, "faulty code".